packages/fcl/src/wallet-utils/encode-provable-authn-signing.js

```diff
@@ -1,20 +1,48 @@
-import {withPrefix} from "@onflow/util-address"
+import {sansPrefix} from "@onflow/util-address"
 import {invariant} from "@onflow/util-invariant"
-import {encode} from '@onflow/rlp'
+import {encode} from "@onflow/rlp"
 
 const rightPaddedHexBuffer = (value, pad) =>
   Buffer.from(value.padEnd(pad * 2, 0), "hex")
 
-export const encodeMessageForProvableAuthnSigning = (address, timestamp, appDomainTag = "") => {
-    invariant(address, "Encode Message From Provable Authn Error: address must be defined")
-    invariant(timestamp, "Encode Message From Provable Authn Error: timestamp must be defined")
+const leftPaddedHexBuffer = (value, pad) =>
+  Buffer.from(value.padStart(pad * 2, 0), "hex")
 
-    const USER_DOMAIN_TAG = rightPaddedHexBuffer(Buffer.from("FLOW-V0.0-user").toString("hex"), 32).toString("hex")
-    const APP_DOMAIN_TAG = rightPaddedHexBuffer(Buffer.from(appDomainTag).toString("hex"), 32).toString("hex")
+const addressBuffer = addr => leftPaddedHexBuffer(addr, 8)
 
-    return USER_DOMAIN_TAG + (appDomainTag ? 
-        encode([APP_DOMAIN_TAG, withPrefix(address), timestamp]).toString("hex")
-        :
-        encode([withPrefix(address), timestamp]).toString("hex")
-    )
+export const encodeMessageForProvableAuthnSigning = (
+  address,
+  timestamp,
+  appDomainTag = ""
+) => {
+  invariant(
+    address,
+    "Encode Message From Provable Authn Error: address must be defined"
+  )
+  invariant(
+    timestamp,
+    "Encode Message From Provable Authn Error: timestamp must be defined"
+  )
+
+  const USER_DOMAIN_TAG = rightPaddedHexBuffer(
+    Buffer.from("FLOW-V0.0-user").toString("hex"),
+    32
+  ).toString("hex")
+
+  const APP_DOMAIN_TAG = appDomainTag
+    ? rightPaddedHexBuffer(Buffer.from(appDomainTag).toString("hex"), 32)
+    : null
+
+  return (
+    USER_DOMAIN_TAG +
+    Buffer.concat([
+      appDomainTag
+        ? encode([
+            APP_DOMAIN_TAG,
+            addressBuffer(sansPrefix(address)),
+            timestamp,
+          ])
+        : encode([addressBuffer(sansPrefix(address)), timestamp]),
+    ]).toString("hex")
+  )
```

Comment:
The call to `Buffer.concat` is redundant here because it only contains one buffer. I would keep `USER_DOMAIN_TAG` as buffer above and then use `concat()` like so:

```suggestion
  const USER_DOMAIN_TAG = rightPaddedHexBuffer(
    Buffer.from("FLOW-V0.0-user").toString("hex"),
    32
  )

  const APP_DOMAIN_TAG = appDomainTag
    ? rightPaddedHexBuffer(Buffer.from(appDomainTag).toString("hex"), 32)
    : null

  return Buffer.concat([
    USER_DOMAIN_TAG,
    appDomainTag
      ? encode([
          APP_DOMAIN_TAG,
          addressBuffer(sansPrefix(address)),
          timestamp,
        ])
      : encode([addressBuffer(sansPrefix(address)), timestamp]),
  ]).toString("hex")
```